src/Pages/Applications/DevtronApps/Details/AppConfigurations/Navigation/EnvConfigurationsNav.tsx

```diff
@@ -0,0 +1,281 @@
+        deploymentTemplate: null,
+        configMaps: [],
+        secrets: [],
+    })
+    const [jobEnvs, setJobEnvs] = useState<AppEnvironment[]>([])
+
+    // CONSTANTS
+    const isDevtronApp = resourceKind === ResourceKindType.devtronApplication
+    /** Current Environment Data. */
+    const environmentData = (isDevtronApp ? environments : jobEnvs)?.find(
+        ({ environmentId }) => environmentId === +envId,
+    ) || {
+        environmentName: 'Base Configurations',
+        environmentId: -1,
+        isProtected: isBaseConfigProtected,
+    }
+
+    const addUnnamedNavLink = (
+        _updatedEnvConfig: ReturnType<typeof getEnvConfiguration>,
+        resourceType: EnvResourceType,
+    ) =>
+        _updatedEnvConfig[resourceType === EnvResourceType.ConfigMap ? 'configMaps' : 'secrets'].push({
+            title: 'Unnamed',
+            href: getNavigationPath(
+                path,
+                appId,
+                environmentData.environmentId,
+                resourceType,
+                ResourceConfigState.Unnamed,
+            ),
+        })
+
+    useEffect(() => {
+        // Fetch the env configuration
+        fetchEnvConfig(+(envId || -1))
+
+        return () => {
+            setExpandedIds(null)
+        }
+    }, [envId])
+
+    useEffect(() => {
+        if (envConfig.config) {
+            const newEnvConfig = getEnvConfiguration(envConfig.config, path, appId, environmentData)
+
+            let resourceType: EnvResourceType
+
+            if (pathname.includes('/configmap')) {
+                resourceType = EnvResourceType.ConfigMap
+            } else if (pathname.includes('/secrets')) {
+                resourceType = EnvResourceType.Secret
+            }
+
+            if (pathname.includes('/create')) {
+                addUnnamedNavLink(newEnvConfig, resourceType)
+            }
+
+            setExpandedIds({ ...expandedIds, [resourceType]: true })
+            setUpdatedEnvConfig(newEnvConfig)
+        }
+    }, [envConfig])
+
+    useEffect(() => {
+        getJobOtherEnvironmentMin(appId)
+            .then(({ result }) => {
+                setJobEnvs(result)
+            })
+            .catch(() => {})
+    }, [])
+
+    // METHODS
+    /** Renders the Deployment Template Nav Icon based on `envConfig`. */
+    const renderDeploymentTemplateNavIcon = () => {
+        const { iconConfig } = updatedEnvConfig.deploymentTemplate
+
+        return iconConfig ? (
+            <Tippy {...iconConfig.tooltipProps}>
+                <div className="flex">
+                    <iconConfig.Icon {...iconConfig.props} className={`icon-dim-16 ${iconConfig.props.className}`} />
+                </div>
+            </Tippy>
+        ) : null
+    }
+
+    /** Handles collapse button click. */
+    const onCollapseBtnClick = (id: string) => {
+        setExpandedIds({ ...expandedIds, [id]: !expandedIds?.[id] })
```

Comment:
Ideally we can update the typing of the expandedIds like Record<number, true> and delete the entry on collapse. This helps in case we need to show the total count of expanded items

Reply:
we don't require to show count of expanded items